src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -100,6 +100,28 @@ end
 
 Base.:*(Aadj::Adjoint{T,<:LazyHcat}, B::AbstractMatrix) where {T} = Aadj * LazyHcat(B)
 
+# Special case of Hermitian result: can only actively compute the block upper diagonal
+@views function mul_hermi(Aadj::Adjoint{T,<:LazyHcat}, B::LazyHcat) where {T}
+    A = Aadj.parent
+    rows = size(A, 2)
+    cols = size(B, 2)
+    ret = similar(B.blocks[1], rows, cols)
+
+    ocol = 0  # column offset
+    for (ib, blB) in enumerate(B.blocks)
+        orow = 0  # row offset
+        for (ia, blA) in enumerate(A.blocks)
+            ib < ia && continue
+            ret[orow .+ (1:size(blA, 2)), ocol .+ (1:size(blB, 2))] .= blA' * blB
+            orow += size(blA, 2)
+        end
+        ocol += size(blB, 2)
+    end
+    Hermitian(ret)
+end
+
+mul_hermi(Aadj::AbstractArray{T}, B::AbstractArray{T}) where {T} = Hermitian(Aadj * B)
```

Comment:
put the generic before the specialization

Reply:
(also no need to type them as abstractarray, or get T just do mul_hermi(A,B) = Hermitian(A*B)